src/frontend/AppContainer.tsx

```diff
@@ -1,5 +1,146 @@
+  : async (navState: NavigationState) => {
+      try {
+        await AsyncStorage.setItem(NAV_STORE_KEY, JSON.stringify(navState));
+      } catch (err) {
+        log("Error saving navigation state", err);
+      }
+    };
+
+const loadNavigationState = IS_E2E
+  ? undefined
+  : async () => {
+      try {
+        const navState = JSON.parse(
+          (await AsyncStorage.getItem(NAV_STORE_KEY)) as string
+        );
+        const didCrashLastOpen = JSON.parse(
+          (await AsyncStorage.getItem(ERROR_STORE_KEY)) as string
+        );
+        // Clear error saved state so that navigation persistence happens on next load
+        await AsyncStorage.setItem(ERROR_STORE_KEY, JSON.stringify(false));
+        // If the app crashed last time, don't restore nav state
+        if (didCrashLastOpen) {
+          bugsnag.leaveBreadcrumb("Crash on last open");
+          log("Crashed on last open, skipping load of navigation state");
+          return null;
+        } else {
+          return navState;
+        }
+      } catch (error) {
+        bugsnag.leaveBreadcrumb("Error loading nav state", { error });
+        log("Error reading navigation and error state", error);
+      }
+    };
+
+const getRouteName = (navState?: NavigationState): string | null => {
+  if (!navState) {
+    return null;
+  }
+
+  const route = navState.routes[navState.index];
+
+  if (route?.routes) {
+    return getRouteName(route);
+  }
+
+  return route.routeName;
+};
+
+const inviteModalDisabledOnRoute = (routeName: string) =>
+  EDITING_SCREEN_NAMES.includes(routeName);
+
+const AppContainer = createAppContainer(WithModalsStack);
+
+const AppContainerWrapper = () => {
+  const navRef = React.useRef<NavigationContainerComponent>();
+  const [inviteModalEnabled, setInviteModalEnabled] = React.useState(true);
+  const [queuedInvite, setQueuedInvite] = React.useState<string | null>(null);
+
+  const onNavStateChange = (
+    previousState: NavigationState,
+    currentState: NavigationState
+  ) => {
+    const previousRouteName = getRouteName(previousState);
+    const currentRouteName = getRouteName(currentState);
+
+    if (previousRouteName !== currentRouteName && currentRouteName) {
+      setInviteModalEnabled(!inviteModalDisabledOnRoute(currentRouteName));
+    }
+  };
+
+  const openInviteModal = React.useCallback((key: string) => {
+    if (navRef.current) {
+      navRef.current.dispatch(
+        NavigationActions.navigate({
+          routeName: "ProjectInviteModal",
+          params: { invite: key },
+        })
+      );
+    }
+  }, []);
+
+  //   useProjectInviteListener(invite => {
+  //     if (inviteModalEnabled) {
+  //       openInviteModal(invite);
+  //     } else {
+  //       setQueuedInvite(invite);
+  //     }
+  //   });
+
+  React.useEffect(() => {
+    if (inviteModalEnabled && queuedInvite) {
+      setQueuedInvite(null);
+      openInviteModal(queuedInvite);
+    }
+  }, [inviteModalEnabled, queuedInvite, openInviteModal]);
```

Comment:
Handles showing the invite modal after the user navigates to a screen that permits it (assuming there's an invite that's been queued). May be worth adding a small delay for the `openInviteModal` call since it can be a little visually jarring for some people